### PR TITLE
Handle alias room permalinks

### PR DESCRIPTION
**Pull request by stefanceriu**
Use the newly exposed RustSDK Client `resolveRoomAlias` method to transform room aliases into permalinks and percolate them through the rets of the application.

### CODE REVIEW

File: ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift [hunk 1]
```diff
@@ -218,25 +189,76 @@ class UserSessionFlowCoordinator: FlowCoordinatorProtocol {
...
+    
     func attemptStartingOnboarding() {
         if onboardingFlowCoordinator.shouldStart {
             clearRoute(animated: false)
             onboardingFlowCoordinator.start()
         }
     }
     
-    private func clearPresentedSheets(animated: Bool, completion: @escaping () -> Void) {
+    private func clearPresentedSheets(animated: Bool) async {
```
Comment (pixlwave) [inline ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift:253]: Nice I tried making this async before but ended up with all the `handleAppRoutes` also being async which was dumb. This approach looks good to me 👍

File: ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift [hunk 2]
```diff
@@ -178,37 +178,8 @@ class UserSessionFlowCoordinator: FlowCoordinatorProtocol {
...
-            case .genericCallLink(let url):
-                navigationSplitCoordinator.setSheetCoordinator(GenericCallLinkCoordinator(parameters: .init(url: url)), animated: animated)
-            case .oidcCallback:
-                break
-            case .settings, .chatBackupSettings:
-                settingsFlowCoordinator.handleAppRoute(appRoute, animated: animated)
-            }
+        Task {
+            await asyncHandleAppRoute(appRoute, animated: animated)
         }
```
Comment (pixlwave) [inline ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift:183]: I wonder if we should do something to make sure only one of these runs at once. Doesn't seem a leap to believe someone might tap a link, app opens on a poor connection, they wait a bit, go back to the link and tap again.

**Review by pixlwave — APPROVED** [review]
LGTM

File: ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift (hunk 2 above)
Comment (stefanceriu) [reply ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift:183]: We do, we block the whole screen 😅

Comment (pixlwave) [reply ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift:183]: That doesn't help with external links 😜

**Pull request merged**